src/theme/index.tsx

```diff
@@ -0,0 +1,22 @@
+import { CssBaseline, createTheme, PaletteMode } from '@mui/material';
+import { ThemeProvider } from '@mui/material/styles';
+import { useMemo } from 'react';
+import { getDesignTokens } from './themeTokens';
+import { useAppSelector } from '../hooks/redux-hooks';
+import { getProfile } from '../stores/slices/authSlice';
+
+const ThemeWrapper = ({ children }: { children: React.ReactNode }) => {
+  const uiTheme = useAppSelector(getProfile)?.uiTheme || localStorage.getItem('ui_theme');
+  const mode: PaletteMode = uiTheme === 'DARK' ? 'dark' : 'light';
+
+  const updatedTheme = useMemo(() => createTheme(getDesignTokens(mode)), [mode]);
```

Comment:
`updatedTheme` why **updated**, just a `theme`

Reply:
You can create two objects with supported themes in advance rather than building in on the fly.

It will improve the readability of your theme object and improve performance a little bit because the theme description object will be created once instead of each time the user switches modes.


```ts
const lightTheme: ThemeOptions = {
  // Declarative description of the light theme
  palette: {
    mode: 'light',
  },
};

const darkTheme: ThemeOptions = {
  // Declarative description of the dark theme
  palette: {
    mode: 'dark',
  },
};

const themeOptions: Record<PaletteMode, ThemeOptions> = {
  light: lightTheme,
  dark: lightTheme,
};

const theme = createTheme(themeOptions[mode])


```